src/NHibernate/Engine/BatchFetchQueue.cs

```diff
@@ -202,9 +266,9 @@ public object[] GetEntityBatch(IEntityPersister persister,object id,int batchSiz
 			int end = -1;
 			bool checkForEnd = false;
 
-			foreach (EntityKey key in batchLoadableEntityKeys.Keys)
+			if (batchLoadableEntityKeys.TryGetValue(persister.EntityName,out var set))//TODO: this needn't exclude subclasses...
```

Comment:
Missing space after comma. And I would rather have the comment on the line above instead of at end of line. I view most end of line comments as bad practice, as they can lack visibility.